### PR TITLE
Fix solc-nightly job

### LINKED ISSUE #1605: Fix solc nightly travis job
Now that we're using Truffle 5 the previous mechanism doesn't work anymore.

https://github.com/OpenZeppelin/openzeppelin-solidity/blob/7fb90a1566d668bea8e25e9c769cf878f14e8ed3/scripts/test.sh#L57-L60

We should be using the Truffle 5 feature to choose the compiler version.

### PR DESCRIPTION
**Pull request by frangio**
Fixes #1605.

Also had to update Truffle to get this to work.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "prepack": "npm run build",
     "release": "scripts/release/release.sh",
     "version": "scripts/release/update-changelog-release-date.js && scripts/release/update-ethpm-version.js",
-    "test": "npm run compile && scripts/test.sh"
+    "test": "scripts/test.sh"
```
Comment (nventuro) [inline package.json]: Why this change? The explicit compile step was there to prevent truffle from recompiling on successive runs of `npm test`. Did the truffle update change this?

File: scripts/test.sh [hunk 2]
```diff
@@ -63,18 +63,17 @@ else
 fi
 
 if [ "$SOLC_NIGHTLY" = true ]; then
-  echo "Downloading solc nightly"
-  wget -q https://raw.githubusercontent.com/ethereum/solc-bin/gh-pages/bin/soljson-nightly.js -O /tmp/soljson.js && find . -name soljson.js -exec cp /tmp/soljson.js {} \;
+  docker pull ethereum/solc:nightly
```
Comment (nventuro) [inline scripts/test.sh]: Is Docker available by default on Travis, or do we have to install it somehow?

File: truffle-config.js [hunk 3]
```diff
@@ -1,5 +1,16 @@
 require('chai/register-should');
 
+const solcStable = {
+  version: '0.5.7',
+};
+
+const solcNightly = {
+  version: 'nightly',
+  docker: true,
```
Comment (nventuro) [inline truffle-config.js]: Is docker a requirement for the nightly build?

File: scripts/test.sh (hunk 2 above)
Comment (frangio) [reply scripts/test.sh]: It's available by default. Note that the job in this PR passed. :slightly_smiling_face:

File: package.json (hunk 1 above)
Comment (frangio) [reply package.json]: This is actually what was breaking the nightly job, because the nightly build is downloaded in `scripts/test.sh`.

Note that is still the case:

https://github.com/OpenZeppelin/openzeppelin-solidity/blob/c0f69c53b8a1cc7b9c13dfe3cd7b5d7e8c7843f6/scripts/test.sh#L65-L67

Truffle is supposed to be doing this automatically, but it doesn't seem to be working.

File: truffle-config.js (hunk 3 above)
Comment (frangio) [reply truffle-config.js]: Yes, it's the only way I found to use a nightly build, they aren't available using solcjs with Truffle.

File: package.json (hunk 1 above)
Comment (nventuro) [reply package.json]: Oh, so it doesn't recompile on compiler changes? Shouldn't that `if ` be in a `compile` script?

File: truffle-config.js (hunk 3 above)
Comment (nventuro) [reply truffle-config.js]: Neat, this shouldn't cause any issues for users anyway.

File: package.json (hunk 1 above)
Comment (frangio) [reply package.json]: Good point...

**Comment by frangio** [comment]
I also moved `scripts/build.sh` to `scripts/prepack.sh` because it was very ambiguous. That (nasty) script is only used to prepare the package.

**Comment by nventuro** [comment]
Did you `chmod +x` the compile script?

**Comment by frangio** [comment]
Yeah, sorry about that.
